spec/light-client/Lightclient.tla

```diff
@@ -128,33 +129,47 @@ OnStart ==
         /\ requestStack' = initStack
         /\ outEvent' = RequestHeaderForTopRequest(initStack)
 
-(* Check whether we can trust the signedHdr based on trustedHdr
-   following the trusting period method.
-   This operator is similar to CheckSupport in the English spec.
+(**
+ Check whether commits in a signed header are correct with respect to the given
+ validator set (DOMAIN votingPower) and votingPower.
+ *)
+Verify(votingPower, signedHdr) ==
+    \* the implementation should check the hashes and other properties of the header
+    LET TP == BC!PowerOfSet(votingPower, DOMAIN votingPower)
+        SP == BC!PowerOfSet(votingPower,
+                            signedHdr[2] \intersect DOMAIN votingPower)
+    IN
+        \* the commits are signed by the validators
+    /\ signedHdr[2] \subseteq DOMAIN signedHdr[1].VP
+        \* the 2/3s rule works
+    /\ 3 * SP > 2 * TP
+
+(* 
+ Check whether we can trust the signedHdr based on trustedHdr
+ following the trusting period method.
+ This operator is similar to CheckSupport in the English spec.
    
-   The parameters have the following meanings:
+ The parameters have the following meanings:
    - heightToTrust is the height of the trusted header
    - heightToVerify is the height of the header to be verified
    - trustedHdr is the trusted header (not a signed header)
    - signedHdr is the signed header to verify (including commits)
-   *)
+ *)
 CheckSupport(heightToTrust, heightToVerify, trustedHdr, signedHdr) ==
     IF minTrustedHeight > heightToTrust \* outside of the trusting period
-        \/ ~(signedHdr[2] \subseteq (DOMAIN signedHdr[1].VP)) \* signed by other nodes
     THEN FALSE
     ELSE
       LET TP == BC!PowerOfSet(trustedHdr.NextVP, DOMAIN trustedHdr.NextVP)
           SP == BC!PowerOfSet(trustedHdr.NextVP,
             signedHdr[2] \intersect DOMAIN trustedHdr.NextVP)
       IN
-      IF heightToVerify = heightToTrust + 1
-          \* the special case of adjacent heights: check 2/3s
-      THEN (3 * SP > 2 * TP)
-      ELSE \* the general case: check 1/3 between the headers and make sure there are 2/3 votes  
+      IF heightToVerify = heightToTrust + 1 \* adjacent headers
+      THEN signedHdr[1].VP = trustedHdr.NextVP
```

Comment:
In case of consecutive headers we still want to check if signedHdr is valid, i.e., that there are more than 2/3 of voting power signed.